### ISSUE TITLE
error: 'ttir.conv_transpose2d' op Number of input channels from input tensor must match the first dimension of the weight tensor. Got X input channels and Y in the weight tensor.

### ISSUE THREAD

**Issue by mmilosevicTT** [tenstorrent/tt-mlir#5784]
### Describe the bug

Several TT-XLA PyTorch models are failing in training with the `ttir.conv_transpose2d` after the fix for batch norm training in https://github.com/tenstorrent/tt-mlir/pull/5287. 

Some of the models failing are:
- SegFormer ([segformer_output.txt](https://github.com/user-attachments/files/23475000/segformer_output.txt))
- EfficientNet ([efficientnet_output.txt](https://github.com/user-attachments/files/23475102/efficientnet_output.txt))
- MobileNetV3 ([mobilenetv3_output.txt](https://github.com/user-attachments/files/23475176/mobilenetv3_output.txt))
- RegNet ([regnet_output.txt](https://github.com/user-attachments/files/23474925/regnet_output.txt))

Extracted problematic ops from outputs are:
- SegFormer (X = 2048, Y = 1)
```
%188 = "ttir.convolution"(%180, %186, %187) <{batch_group_count = 1 : i64, convolution_layout = #ttir<convolution_layout input_batch = 0, input_feature = 1, input_spatial_dimensions = 2x3, kernel_output_feature = 2, kernel_input_feature = 3, kernel_spatial_dimensions = 0x1, output_batch = 0, output_feature = 1, output_spatial_dimensions = 2x3>, feature_group_count = 2048 : i64, input_dilation = array<i64: 1, 1>, padding = array<i64: 1, 1, 1, 1>, weight_dilation = array<i64: 1, 1>, window_reversal = array<i1: false, false>, window_strides = array<i64: 1, 1>}> : (tensor<1x2048x16x16xbf16>, tensor<3x3x2048x1xbf16>, tensor<1x2048x16x16xbf16>) -> tensor<1x2048x16x16xbf16> loc(#loc387)
```
- EfficientNet (X = 1152, Y = 1)
```
%328 = "ttir.convolution"(%302, %326, %327) <{batch_group_count = 1 : i64, convolution_layout = #ttir<convolution_layout input_batch = 0, input_feature = 1, input_spatial_dimensions = 2x3, kernel_output_feature = 2, kernel_input_feature = 3, kernel_spatial_dimensions = 0x1, output_batch = 0, output_feature = 1, output_spatial_dimensions = 2x3>, feature_group_count = 1152 : i64, input_dilation = array<i64: 1, 1>, padding = array<i64: 1, 1, 1, 1>, weight_dilation = array<i64: 1, 1>, window_reversal = array<i1: false, false>, window_strides = array<i64: 1, 1>}> : (tensor<1x1152x7x7xbf16>, tensor<3x3x1152x1xbf16>, tensor<1x1152x7x7xbf16>) -> tensor<1x1152x7x7xbf16> loc(#loc371)
```
- MobileNetV3 (X = 576, Y = 1)
```
%471 = "ttir.convolution"(%449, %469, %470) <{batch_group_count = 1 : i64, convolution_layout = #ttir<convolution_layout input_batch = 0, input_feature = 1, input_spatial_dimensions = 2x3, kernel_output_feature = 2, kernel_input_feature = 3, kernel_spatial_dimensions = 0x1, output_batch = 0, output_feature = 1, output_spatial_dimensions = 2x3>, feature_group_count = 576 : i64, input_dilation = array<i64: 1, 1>, padding = array<i64: 2, 2, 2, 2>, weight_dilation = array<i64: 1, 1>, window_reversal = array<i1: false, false>, window_strides = array<i64: 1, 1>}> : (tensor<1x576x7x7xbf16>, tensor<5x5x576x1xbf16>, tensor<1x576x7x7xbf16>) -> tensor<1x576x7x7xbf16> loc(#loc375)
```
- RegNet (X = 1088, Y = 64)
```
%267 = "ttir.convolution"(%241, %265, %266) <{batch_group_count = 1 : i64, convolution_layout = #ttir<convolution_layout input_batch = 0, input_feature = 1, input_spatial_dimensions = 2x3, kernel_output_feature = 2, kernel_input_feature = 3, kernel_spatial_dimensions = 0x1, output_batch = 0, output_feature = 1, output_spatial_dimensions = 2x3>, feature_group_count = 17 : i64, input_dilation = array<i64: 1, 1>, padding = array<i64: 1, 1, 1, 1>, weight_dilation = array<i64: 1, 1>, window_reversal = array<i1: false, false>, window_strides = array<i64: 1, 1>}> : (tensor<1x1088x7x7xbf16>, tensor<3x3x1088x64xbf16>, tensor<1x1088x7x7xbf16>) -> tensor<1x1088x7x7xbf16> loc(#loc609)
```
Outputs of some versions of these models are provided above.

### Steps to reproduce the issue

1. Go to TT-XLA
2. Replace NOT_SUPPORTED_SKIP tag with EXPECTED_PASSING in the `tests/runner/test_config/torch/test_config_training_single_device.yaml`
3. Run `pytest tests/runner/test_models.py::test_all_models_torch[<TEST_NAME>]`, or any other of the mentioned models.
4. Full list of models failing with this error is present in `tests/runner/test_config/torch/test_config_training_single_device.yaml`

**Comment by mmilosevicTT** [comment]
@sdjordjevicTT @djstefanovicTT @vladimirjovanovicTT Pinging for visibility.

**Comment by vladimirjovanovicTT** [comment]
@mmilosevicTT can you extract the shlo ops in question from the whole graphs and add them in issue body?

**Comment by mmilosevicTT** [comment]
Added pointer to the op that was problematic.
